m3u8/writer.go

```diff
@@ -297,6 +301,81 @@ func writeExtXIFrameStreamInf(buf *bytes.Buffer, vnt *Variant) {
 	buf.WriteRune('\n')
 }
 
+func writePartialSegment(buf *bytes.Buffer, ps *PartialSegment) {
+	if !ps.ProgramDateTime.IsZero() {
+		buf.WriteString("#EXT-X-PROGRAM-DATE-TIME:")
+		buf.WriteString(ps.ProgramDateTime.Format(DATETIME))
+		buf.WriteRune('\n')
+	}
+	buf.WriteString("#EXT-X-PART:")
+	buf.WriteString("DURATION=")
+	buf.WriteString(strconv.FormatFloat(ps.Duration, 'f', 3, 64))
+	if ps.Independent {
+		buf.WriteString(",INDEPENDENT=YES")
+	}
+	if ps.Gap {
+		buf.WriteString(",GAP=")
+		writeYESorNO(buf, ps.Gap)
+	}
+	if ps.Limit > 0 {
+		buf.WriteString(",BYTERANGE=")
+		buf.WriteString(strconv.FormatInt(ps.Limit, 10))
+		buf.WriteRune('@')
+		buf.WriteString(strconv.FormatInt(ps.Offset, 10))
+	}
+	buf.WriteString(",URI=\"")
+	buf.WriteString(ps.URI)
+	buf.WriteRune('"')
+	buf.WriteRune('\n')
+}
+
+func writePreloadHint(buf *bytes.Buffer, ph *PreloadHint) {
+	buf.WriteString("#EXT-X-PRELOAD-HINT:")
+	buf.WriteString("TYPE=")
+	buf.WriteString(ph.Type)
+	buf.WriteString(",URI=\"")
+	buf.WriteString(ph.URI)
+	buf.WriteRune('"')
+	if ph.Limit > 0 {
+		buf.WriteString(",BYTERANGE-START=")
+		buf.WriteString(strconv.FormatInt(ph.Offset, 10))
+		buf.WriteString(",BYTERANGE-LENGTH=")
+		buf.WriteString(strconv.FormatInt(ph.Limit, 10))
+	}
+	buf.WriteRune('\n')
+}
+
+func writeSkip(buf *bytes.Buffer, skippedSegments uint64) {
+	buf.WriteString("#EXT-X-SKIP:")
+	buf.WriteString("SKIPPED-SEGMENTS=")
+	buf.WriteString(strconv.FormatUint(skippedSegments, 10))
+	buf.WriteRune('\n')
+}
+
+func writeServerControl(buf *bytes.Buffer, sc *ServerControl) {
+	buf.WriteString("#EXT-X-SERVER-CONTROL:")
+	stringsToWrite := []string{}
```

Comment:
A more efficient way to build strings is to use strings.Builder: https://brandur.org/fragments/bytes-buffer-vs-strings-builder. You don't need to change the code, though.